<a href="https://colab.research.google.com/github/ADRIANVM117/proyectos_propios/blob/main/Estadistica/estadistica_ejercicio5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Tarea 28-Ene-21

- 1) El gerente de la división de bombillas de la Cardinal Electric debe estimar el número promedio de horas que durarán los focos fabricados por cada una de las máquinas. Fue elegida una muestra de 40 focos de la máquina A y el tiempo promedio de funcionamiento fue de 1,416 horas. Se sabe que la desviación estándar poblacional de la duración es de 30 horas.
  - a) Calcule el error estándar de la media.
  - b) Construya un intervalo de confianza con 90%, 95%, 98% y 99% para la media de la población.


## Datos.
   
   ### $ n = 40 $
   ### $ \mu = 1416 $
   ### $ \sigma = 30 $
  


In [ ]:
import statistics
import numpy as np 
from scipy import stats
n = 40 # tamaño muestra
mu = 1416 #media 
sigma = 30 # desviación poblacional 

error_est = sigma / np.sqrt(n)
NC = .99 # Nivel de confianza de 99% 
N_colas = 2 # Numero de colas 
alpha = (1-NC)/2

z_score = stats.norm.ppf(1-alpha)


print('Error Estandar : \t{:.3f}'.format(error_est))
print('z-score : \t{:.3f}'.format(z_score))

Error Estandar : 	4.743
z-score : 	2.576


# Intervalos de confiaza para 90%, 95%, 98% y 99%

In [ ]:
confianza = [.90, .95, .98, .99]

for i in confianza: 
  alpha1 = (1- i)/ 2 
  z_score1 = stats.norm.ppf(1-alpha1)
  margen_error = z_score1 * error_est
  li = mu - margen_error
  ls = mu + margen_error
  print('Limite inferior : \t{:.3f}. Limite Superior : \t{:.3f}. Para un nivel de confianza de : \t{} '.format(li,ls,i))

Limite inferior : 	1408.198. Limite Superior : 	1423.802. Para un nivel de confianza de : 	0.9 
Limite inferior : 	1406.703. Limite Superior : 	1425.297. Para un nivel de confianza de : 	0.95 
Limite inferior : 	1404.965. Limite Superior : 	1427.035. Para un nivel de confianza de : 	0.98 
Limite inferior : 	1403.782. Limite Superior : 	1428.218. Para un nivel de confianza de : 	0.99 


# Conclusión. 

- Para un nivel de confianza del 90%
  - El intervalo de confianza se encuentra entre 1408.198 y 1423.802 horas. 

- Para un nivel de confianza del 95%
  - El intervalo de confianza se encuentra entre 1406.703 y 1425.297 horas. 

- Para un nivel de confianza del 98%
  - El intervalo de confianza se encuentra entre 1404.965 y 1427.035 horas. 

- Para un nivel de confianza del 99%
  - El intervalo de confianza se encuentra entre 1403.782 y 1428.218 horas. 


- 2) Analizando el comportamiento de una muestra de 25 acciones que cotizan en la BMV, se encontró la variación promedio diaria fue del 16% con una desviación estándar del 4%. Estime, con un 95% de nivel de confianza, la variación porcentual promedio diaria que tienen las acciones que cotizan en la Bolsa.

#### Datos. 

### $ n = 25 $
### $ \mu = 16 $
### $ \sigma = 4$
### Nivel de confianza del 95% 






## Como nos encontramos con un problema de una muestra menor a 30. Usaremos la distribución t-student para resolverlo. 

In [ ]:
n = 25 # muestra 
mu = 16 
sigma = 4 
NC = .95 
alpha = (1-NC)/ 2
t_student = stats.t.ppf(q= 1- alpha, df=24, loc=0, scale=1)
error_est = sigma / np.sqrt(n)

print('Valor t : \t{:.3f}'.format(t_student))
print('Error Estandar : \t{:.3f}'.format(error_est))



Valor t : 	2.064
Error Estandar : 	0.800


Siempre que realice una prueba de dos colas, habrá dos valores críticos. En este caso, los valores críticos de T son 2.064 y -2.64 . Por lo tanto, si el estadístico de la prueba es menor que -2.064 o mayor que 2.064, los resultados de la prueba son estadísticamente significativos.

# Intervalos de confianza

## Margen de error. 
   - $ ME = t_{-student} * \sigma_{x} $

In [ ]:
ME = t_student * error_est
li = mu - ME
ls = mu + ME

print('Limite Inferior \t{:.3f}'.format(li))
print('Limite Superior \t{:.3f}'.format(ls))

Limite Inferior 	14.349
Limite Superior 	17.651


# Conclusión. 

## Con un nivel de confianza del 95% la variación porcentual de las acciones se encuentra ente 14.349 y 17.651 

# 3)
## Se desea encontrar un intervalo de los ingresos semanales de los alumnos de una universidad. Para este estudio se encuentra los siguientes datos y la población de la universidad que es de 3,500 alumnos.

In [ ]:
datos = [4000,	500,	1500,	500	,2500	,1500,
         1500, 6000	,500	,1000	,1200	,1500,
         500	,1500	,1500	,800	,1000	,2500
         ,1000	,2500	,1500	,3500	,500	
         ,500	,2500	,500	,2000	,2500]

In [ ]:
datos = np.array(datos)
N_b= 3500 
n_b= len(datos)

print('Tamaño poblacional : \t{}'.format(N_b))
print('Tamaño Muestral : \t{}'.format(n_b))

Tamaño poblacional : 	3500
Tamaño Muestral : 	28


In [ ]:
mu_pob = np.mean(datos)
sigma_b = 1265

print('Media Poblacional : \t{:.2f}'.format(mu_pob))
print('Desviación Est. Poblacional : \t{:.2f}'.format(sigma_b))

Media Poblacional : 	1678.57
Desviación Est. Poblacional : 	1265.00


## Error estandar de la media de una población finita 


## $ \bar{\sigma_{x}} = \frac{\sigma}{\sqrt{n}} * \sqrt{\frac{N -n}{N-1}}$

In [ ]:
Error_est_med = (sigma_b / np.sqrt(n_b)) * (np.sqrt( (N_b-n_b)/(N_b-1 )  ))
print('Error estandar de la media de una población finita: \t{:.2f}'.format(Error_est_med))

Error estandar de la media de una población finita: 	238.14


In [ ]:
NC = .90
alpha = (1-NC)/ 2
t_student = stats.t.ppf(q= 1- alpha, df=27, loc=0, scale=1)
error_est = sigma / np.sqrt(n)
print('t-student: \t{:.3f}'.format(t_student))

t-student: 	1.703


## Margen de Error 

ME = $ \bar{\sigma_{x}} * t_{-student} $

In [ ]:
ME_b = t_student * Error_est_med
print('Margen de error: \t{:.3f}'.format(ME))

Margen de error: 	1.651


In [ ]:
li =  mu_pob - ME_b
ls =  mu_pob + ME_b
print('Limite inferior: \t{:.3f}'.format(li))
print('Limite Superior: \t{:.3f}'.format(ls))

Limite inferior: 	1272.953
Limite Superior: 	2084.190
